pkg/sourcer/interface.go

```diff
@@ -18,6 +18,8 @@ type Sourcer interface {
 	// When the return value is negative, it indicates the pending information is not available.
 	// With pending information being not available, the Numaflow platform doesn't auto-scale the source.
 	Pending(ctx context.Context) int64
+	//Partitions returns the partitions associated with the source.
```

Comment:
More comments are appreciated. It's really hard for someone who wants to write a UDSource to understand how he/she should implement this function.

Reply:
+1